scripts/pre-commit

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+
+import re
+import sys
+
+import lib.format
+import pylava.config
+import pylava.core
+import pylava.main
+from lib import REPO_ROOT, CheckError, MultiException, run
+from lib.bump_version import bump_with_version, get_next_patch
+
+CHART_VERSION_NOT_OKAY_REGEX = re.compile(
+    r'^.*path: "(.*)".*Chart version.*$', re.MULTILINE
+)
+
+
+def run_chart_lint() -> list[str]:
+    lint_result = run("ct lint --remote upstream".split(), capture_stdout=True)
+    print(lint_result.stdout)
+    print()
+
+    if lint_result.returncode != 0:
+        return CHART_VERSION_NOT_OKAY_REGEX.findall(lint_result.stdout)
+    else:
+        return []
+
+
+def chart_lint():
+    print("🚢 Linting helm charts")
+    changed_charts = set()
+    failed_charts = run_chart_lint()
+    while failed_charts:
+        for chart in failed_charts:
+            next_patch = get_next_patch(chart)
+            bump_with_version(chart, next_patch)
+            changed_charts.add(chart)
+
+            print(f"Updated chart {chart} version to {next_patch}")
+
+        failed_charts = run_chart_lint()
+        if failed_charts:
+            print("🚢 Linting helm charts again")
+
+    if changed_charts:
+        raise CheckError("ct lint", str(changed_charts))
+
+
+def format():
+    print("🐍 Checking python formatting")
+    lib.format.main(check=True)
+
+
+def pylava_lint():
+    print("🐍 🌋 Linting with pylava")
+    pylava_conf = pylava.config.parse_options(sys.argv[1:])
+    pylava.config.setup_logger(pylava_conf)
+    errs = pylava.main.check_path(pylava_conf, rootdir=REPO_ROOT)
+    if errs:
+        formatted_errs = "\n".join(
+            "%(filename)s:%(lnum)s:%(col)s: [%(type)s] %(text)s" % err for err in errs
+        )
+        raise CheckError("pylava", formatted_errs)
+
+
+if __name__ == "__main__":
```

Comment:
I like this idea of showing multiple errors at once, but it's really unusual one. If this was a single tool with multiple command line options invoked by pre-commit hook it would be more convenient to use even outside of pre-commit hooks in parralel invokations, e,g it will be hard now to do:
make chart-lint
make chart-format
make pylava-lint

the above could be entry points to the script that uses exactly same logic as your here, just a bit separated from git pre-commit, so as example:
```
scripts/chart-formatting.py
python scripts/chart-formatting.py --formatter pylava,chart-lint 
```

Reply:
While multiple errors at once is an unusual one, I think it's important. I think this is somewhere where the unix philosophy of "fail fast" is at odds with good UX.

Your makefile example is a really, really good point however. This could be heavily simplified with that.

I'm going to switch this to a WIP PR until I've investigated that.

Reply:
For described is a nice one and I think still easily achievable with the single command and error from it, simple run will continue if the error happens on the pylava first formatter, but will allow separate different formatting / testing frameworks in the future:

`#!/usr/bin/env bash
set +e
python3 scripts/chart-formatting.py --formatter pylava
python3 scripts/chart-formatting.py --formatter chart-lint
`

Reply:
The one thing that script doesn't have is exiting with failure at the end if any task fails. That would be necessary for a pre-commit hook, for example. But that wouldn't be too bad to add in bash.